onnxruntime/contrib_ops/webgpu/bert/attention.cc

```diff
@@ -436,7 +436,7 @@ Status ApplyAttention(const Tensor* Q, const Tensor* K, const Tensor* V, const T
                       WebgpuAttentionParameters& parameters, onnxruntime::webgpu::ComputeContext& context, const Tensor* seqlen_k) {
   const int output_count = std::min({context.OutputCount(), 1 + (past_key != nullptr ? 1 : 0) + (past_value != nullptr ? 1 : 0)});
   const int past_sequence_length = output_count > 1 ? parameters.past_sequence_length_ : 0;
-  const int total_sequence_length = past_sequence_length + parameters.kv_sequence_length_;
+  const int total_sequence_length = parameters.is_gqa_ && parameters.past_present_share_buffer_ ? parameters.seqlen_present_kv_cache_ : (past_sequence_length + parameters.kv_sequence_length_);
```

Comment:
For GQA, total_sequence_length is read from node input like
https://github.com/microsoft/onnxruntime/blob/e12421be3054bdbeefa974a51bd2adf836016871/onnxruntime/contrib_ops/cpu/bert/group_query_attention_helper.h#L183

seqlen_present_kv_cache is the max buffer length, when past and present share buffer.

Reply:
[Here](https://github.com/microsoft/onnxruntime/blob/b14b4ec70382a57faf2d663afc3ca1936bfd20de/onnxruntime/contrib_ops/cpu/bert/group_query_attention_helper.h#L183C1-L184C87) we are assigning present_sequence_length, which is seqlen_present_kv_cache in GQA parameters. The WebGPU implementation uses CheckInputs implementation in onnxruntime\contrib_ops\cpu\bert\group_query_attention_helper.h

Reply:
The use of the name `total_sequence_length` is generalized in that sense that it corresponds to the present key and present value sequence_lengths.

Reply:
that does not make sense: total_sequence_length  is not total_sequence_length  ?

https://github.com/microsoft/onnxruntime/blob/main/onnxruntime/contrib_ops/cpu/bert/group_query_attention_helper.h#L183